testbench/common.py

```diff
@@ -311,6 +311,205 @@ def extract_media(request):
     return request.data
 
 
+def make_json_preconditions(request, prefix="if"):
+    """Create the pre-conditions for most JSON requests.
+
+    The prefix parameter can be `if` or `ifSource` to handle the special names
+    used for source pre-conditions in `Objects: copy` and `Objects: rewrite`.
+    """
+
+    def if_metageneration_match(blob, _, ctx):
+        match = request.args.get(prefix + "MetagenerationMatch", None)
+        assert match is not None
```

Comment:
Why have this assertion separately? I think if you skip the default `None` in the line above it will just throw a KeyError there.

Reply:
I must admit I was just cut&pasting previous code I have written here, but I think this is correct. `flask.Request` returns `None` by default:

https://flask.palletsprojects.com/en/2.0.x/api/#flask.Request.args

https://werkzeug.palletsprojects.com/en/2.0.x/datastructures/#werkzeug.datastructures.MultiDict.get

By construction the `None` value is not possible, an assertion is mostly to make the code easier to read / follow.